metabase/metabase
# Fix x-ray dashboards run some queries twice on load

Closes #36775 

X-ray code has a [check](https://github.com/metabase/metabase/blob/8a9a0a162bdc7c992d059175fd9d5df9784a1d51/frontend/src/metabase/dashboard/hoc/DashboardData.jsx#L95) that should rerun queries when parameter values changes. Apparently, the first time it's triggered, it receives two maps: one is just empty (`{}`), and another has all the parameter values set to `null`. Even though they both basically mean "dashboard filters don't have values set", underscore's `isEqual` treats them as different, and that triggers extra query reruns.

This PR is rather a band-aid, we should invest some time into unifying this logic with regular and public/embedded dashboards that don't have this issue in the first place

### How to verify

1. Open "Browse data", pick any table and x-ray it (click the bolt icon)
2. Ensure there're no canceled `/api/dataset` requests in the Network tab
